pkg/reconciler/pipelinerun/resources/pipelinerunresolution_test.go

```diff
@@ -983,6 +984,196 @@ func TestIsSkipped(t *testing.T) {
+		}, {
+			// skipped because parent was skipped and when expressions are scoped to branch
+			PipelineTask: &v1beta1.PipelineTask{
+				Name:     "mytask18",
+				TaskRef:  &v1beta1.TaskRef{Name: "task"},
+				RunAfter: []string{"mytask11"},
+			},
+			TaskRunName: "pipelinerun-ordering-dependent-task-1",
+			TaskRun:     nil,
+			ResolvedTaskResources: &resources.ResolvedTaskResources{
+				TaskSpec: &task.Spec,
+			},
+		}},
+		scopeWhenExpressionsToTask: false,
+		expected: map[string]bool{
+			"mytask11": true,
+			"mytask18": true,
+		},
+	}, {
+		name: "tasks-when-expressions-scoped-to-task",
+		state: PipelineRunState{{
+			// skipped because when expressions evaluate to false
+			PipelineTask: &pts[10],
+			TaskRunName:  "pipelinerun-guardedtask",
+			TaskRun:      nil,
+			ResolvedTaskResources: &resources.ResolvedTaskResources{
+				TaskSpec: &task.Spec,
+			},
+		}, {
+			// not skipped regardless of its parent task being skipped because when expressions are scoped to task
+			PipelineTask: &v1beta1.PipelineTask{
+				Name:     "mytask18",
+				TaskRef:  &v1beta1.TaskRef{Name: "task"},
+				RunAfter: []string{"mytask11"},
+			},
+			TaskRunName: "pipelinerun-ordering-dependent-task-1",
+			TaskRun:     nil,
+			ResolvedTaskResources: &resources.ResolvedTaskResources{
+				TaskSpec: &task.Spec,
+			},
+		}},
+		scopeWhenExpressionsToTask: true,
+		expected: map[string]bool{
+			"mytask11": true,
+			"mytask18": false,
+		},
+	}, {
+		name: "tasks-when-expressions-scoped-to-branch-skip-multiple-dependent-tasks",
+		state: PipelineRunState{{
+			// skipped because when expressions evaluate to false
+			PipelineTask: &pts[10],
+			TaskRunName:  "pipelinerun-guardedtask",
+			TaskRun:      nil,
+			ResolvedTaskResources: &resources.ResolvedTaskResources{
+				TaskSpec: &task.Spec,
+			},
+		}, {
+			// skipped because parent was skipped and when expressions are scoped to branch
+			PipelineTask: &v1beta1.PipelineTask{
+				Name:     "mytask18",
+				TaskRef:  &v1beta1.TaskRef{Name: "task"},
+				RunAfter: []string{"mytask11"},
+			},
+			TaskRunName: "pipelinerun-ordering-dependent-task-1",
+			TaskRun:     nil,
+			ResolvedTaskResources: &resources.ResolvedTaskResources{
+				TaskSpec: &task.Spec,
+			},
+		}, {
+			// skipped because parent was skipped and when expressions are scoped to branch
+			PipelineTask: &v1beta1.PipelineTask{
+				Name:     "mytask19",
+				TaskRef:  &v1beta1.TaskRef{Name: "task"},
+				RunAfter: []string{"mytask18"},
+			},
+			TaskRunName: "pipelinerun-ordering-dependent-task-2",
+			TaskRun:     nil,
+			ResolvedTaskResources: &resources.ResolvedTaskResources{
+				TaskSpec: &task.Spec,
+			},
+		}},
+		scopeWhenExpressionsToTask: false,
+		expected: map[string]bool{
+			"mytask11": true,
+			"mytask18": true,
+			"mytask19": true,
+		},
+	}, {
+		name: "tasks-when-expressions-scoped-to-task-run-multiple-dependent-tasks",
+		state: PipelineRunState{{
+			// skipped because when expressions evaluate to false
```

Comment:
thanks for adding these comments, really helps me understand the tests! :D

Reply:
glad to hear it 😁